### PR TITLE
added method isChatting

### PR DESCRIPTION
**Pull request by gabrielperales**
Hi @fbartho 

I have added the method `isChatting`. I'm not good at testing this library, actually, I don't know how to do it properly, so I don't know if it is working 😓 . Can you have a look and give me some direction?

Thanks.

### CODE REVIEW

**Comment by gabrielperales** [comment]
@fbartho tested on iOS and it works :)

File: android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java [hunk 1]
```diff
@@ -377,4 +377,18 @@ public void update(ChatState chatState) {
             }
         });
     }
+
+	@ReactMethod
+	public void isChatting(Promise promise) {
+        // Create a temporary observation scope until we get the chat state
+		observationScope = new ObservationScope();
```
[the remark this answers is not in the document]
Comment (gabrielperales) [reply android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java:384]: Yeah, java must allow you to do that. And I'm with you, but I was just _copy pasting_ the structure of the code above :)

[the remark this answers is not in the document]
Comment (fbartho) [reply android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java:384]: Right, but if only 1 thing was using an instance variable then there's no problems, but if 2 things are sharing an instance variable, we have to see if their lifecycles might overlap, if both API calls can exist at the same time then there could be a conflict, where calling the second one cancels the first?

**Comment by fbartho** [comment]
After further thinking, this feels like it's not so much intended to be a Single Async (Promise) Boolean check, but rather an EventEmitter that notifies the JS side persistently.

Are you sure you want this to just be a promise?

**Comment by gabrielperales** [comment]
For my use case, I just need to check whether we have an ongoing chat before opening a new one, so to me is ok to be a promise with a boolean check. What do you suggest? change it for a callback?

**Pull request closed without merging**